storage/sectorstorage/stores/local.go

```diff
@@ -309,6 +309,64 @@ func (st *Local) Remove(ctx context.Context, sid abi.SectorID, typ sectorbuilder
 	return nil
 }
 
+func (st *Local) MoveStorage(ctx context.Context, s abi.SectorID, types sectorbuilder.SectorFileType) error {
+	dest, destIds, sdone, err := st.AcquireSector(ctx, s, 0, types, false)
+	if err != nil {
+		return xerrors.Errorf("acquire dest storage: %w", err)
+	}
+	defer sdone()
+
+	src, srcIds, ddone, err := st.AcquireSector(ctx, s, types, 0, false)
+	if err != nil {
+		return xerrors.Errorf("acquire src storage: %w", err)
+	}
+	defer ddone()
+
+	for _, fileType := range pathTypes {
+		if fileType&types == 0 {
+			continue
+		}
+
+		sst, err := st.index.StorageInfo(ctx, ID(sectorutil.PathByType(srcIds, fileType)))
+		if err != nil {
+			return xerrors.Errorf("failed to get source storage info: %w", err)
+		}
+
+		dst, err := st.index.StorageInfo(ctx, ID(sectorutil.PathByType(destIds, fileType)))
+		if err != nil {
+			return xerrors.Errorf("failed to get source storage info: %w", err)
+		}
+
+
+		if sst.ID == dst.ID {
+			log.Debugf("not moving %v(%d); src and dest are the same", s, fileType)
+			continue
+		}
+
+		if sst.CanStore {
+			log.Debugf("not moving %v(%d); source supports storage", s, fileType)
+			continue
+		}
+
+		log.Debugf("moving %v(%d) to storage: %s(se:%t; st:%t) -> %s(se:%t; st:%t)", s, fileType, sst.ID, sst.CanSeal, sst.CanStore, dst.ID, dst.CanSeal, dst.CanStore)
+
+		if err := st.index.StorageDropSector(ctx, ID(sectorutil.PathByType(srcIds, fileType)), s, fileType); err != nil {
```

Comment:
maybe do the drop after the move completes successfully? Ideally we make it atomic somehow and handle the edge cases properly.

not blocking though

Reply:
Then we would have to guarantee it's readable while it's being moved (not that there's anything that would be reading it)